Helper/CaseManagement.php

```diff
@@ -0,0 +1,184 @@
+ *  ############   ############  #############   ############  #####  ######
+ *                                      ######
+ *                               #############
+ *                               ############
+ *
+ * Adyen Payment module (https://www.adyen.com/)
+ *
+ * Copyright (c) 2021 Adyen BV (https://www.adyen.com/)
+ * See LICENSE.txt for license details.
+ *
+ * Author: Adyen <magento@adyen.com>
+ */
+
+namespace Adyen\Payment\Helper;
+
+use Adyen\Payment\Logger\AdyenLogger;
+use Magento\Framework\App\Helper\AbstractHelper;
+use Magento\Framework\App\Helper\Context;
+use Magento\Sales\Model\Order;
+
+/**
+ * Helper class for anything related to Case Management (Manual Review)
+ *
+ * Class AdyenOrderPayment
+ * @package Adyen\Payment\Helper
+ */
+class CaseManagement extends AbstractHelper
+{
+    const FRAUD_MANUAL_REVIEW = 'fraudManualReview';
+
+    /**
+     * @var AdyenLogger
+     */
+    private $adyenLogger;
+
+    /**
+     * @var Config
+     */
+    private $configHelper;
+
+    /**
+     * CaseManagement constructor.
+     *
+     * @param Context $context
+     * @param AdyenLogger $adyenLogger
+     */
+    public function __construct(
+        Context $context,
+        AdyenLogger $adyenLogger,
+        Config $configHelper
+    ) {
+        parent::__construct($context);
+        $this->adyenLogger = $adyenLogger;
+        $this->configHelper = $configHelper;
+    }
+
+    /**
+     * Based on the passed array, check if manual review is required
+     *
+     * @param array $additionalData
+     * @return bool
+     */
+    public function requiresManualReview(array $additionalData): bool
+    {
+        if (!array_key_exists(self::FRAUD_MANUAL_REVIEW, $additionalData)) {
+            return false;
+        }
+
+        // Strict comparison to 'true' since it will be sent as a string
+        if ($additionalData[self::FRAUD_MANUAL_REVIEW] === 'true') {
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * Mark an order as pending manual review by adding a comment and also, update the status if the review status is set.
+     *
+     * @param Order $order
+     * @param string $pspReference
+     * @param bool $autoCapture
+     * @return Order
+     */
+    public function markCaseAsPendingReview(Order $order, string $pspReference, bool $autoCapture = false): Order
+    {
+        $manualReviewComment = sprintf(
+            'Manual review required for order w/pspReference: %s. Please check the Adyen platform.',
+            $pspReference
+        );
+
+        $reviewRequiredStatus = $this->configHelper->getFraudStatus(
+            Config::XML_STATUS_FRAUD_MANUAL_REVIEW,
+            $order->getStoreId()
+        );
+
+        // Set is in process to false since on Auto Capture, the payment would have already gone trough on the Adyen
+        // platform. The invoice should also be already generated.
+        if ($autoCapture) {
+            $order->setIsInProcess(false);
```

Comment:
Hi @mirkocesaro 

I've tested it on my end and the order is actually set to the `new` state, once the Authorization w/case management webhook is processed. If the `$order->setIsInProcess(false);` line is omitted, the order won't be set to the `new` state because of this [if condition](https://github.com/magento/magento2/blob/3d4095f9f4e691e82d002aedc5a228de1128cd1d/app/code/Magento/Sales/Model/ResourceModel/Order/Handler/State.php#L27).

I've also attached a screenshot showing two orders:
* On order **48** there are no modifications to the PR
* On order **49** I commented out the `$order->setIsInProcess(false);`

![Screen Shot 2021-12-08 at 9 20 25 AM](https://user-images.githubusercontent.com/11156828/145175245-aab189ec-84bc-4d84-9d6b-6414dbe04ad6.png)

Reply:
Hi @Morerice ,
as you noticed, when the `payment_pre_authorized` status with `processing` state is configured, the result is not correct. Of course, the status of the manual review is set correctly set but the wrong state shows a wrong set of status in the update from the admin. This is exactly the problem I found with my configuration. 

And this is the reason I think _setPrePaymentAuthorized and _prepareInvoice methods should be reviewed to work together.